cabal-install/Distribution/Solver/Modular/Log.hs

```diff
@@ -72,17 +77,21 @@ logToProgress mbj l = let
     -- exhaustiveness and first conflict set.
     go :: Progress Message a b
        -> Progress Message a b
-       -> Progress String (Bool, Maybe (ConflictSet QPN)) b
+       -> Progress String (Exhaustive, ConflictSet QPN, ConflictMap) b
        -> Progress String String b
     go ms (Step _ ns) (Step x xs)           = Step x (go ms ns xs)
     go ms r           (Step x xs)           = Step x (go ms r  xs)
-    go ms _           (Fail (exh, Just cs)) = Fail $
-                                              "Could not resolve dependencies:\n" ++
-                                              unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
-                                              (if exh then "Dependency tree exhaustively searched.\n"
-                                                      else "Backjump limit reached (" ++ currlimit mbj ++
-                                                               "change with --max-backjumps or try to run with --reorder-goals).\n")
-                                                  where currlimit (Just n) = "currently " ++ show n ++ ", "
-                                                        currlimit Nothing  = ""
+    go ms _           (Fail (exh, cs, cm))  = Fail $
+        "Could not resolve dependencies:\n" ++
+        unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
```

Comment:
The solver always prints `ms`, which is the log to the first error.  It truncates the log on the line `es = proc (Just 0) l -- catch first error (always)` above.

> BTW, what is the original rationale anyway for showing the first rather than the last error in case the backjump limit is reached?

According to the discussion in #941, the first error was chosen because the solver has made fewer compromises at that point, and the conflicts are more likely to be relevant.  Also, I don't think that there is an obvious place to start printing the summarized log leading up to the final error.  The log wouldn't always include the top of the tree.